Runtime/Data/Collection/ObservableList.cs

```diff
@@ -16,9 +18,27 @@ public abstract class ObservableList<TElement, TEvent> : MonoBehaviour where TEv
         /// <summary>
         /// The collection to observe changes of.
         /// </summary>
-        [SerializeField, DocumentedByXml]
-        protected List<TElement> elements = new List<TElement>();
+        [Serialized]
+        [field: DocumentedByXml]
+        protected List<TElement> Elements { get; set; } = new List<TElement>();
 
+        /// <summary>
+        /// The pointer for the current index being specified in the collection.
+        /// </summary>
+        [Serialized]
+        [field: DocumentedByXml]
+        public int CurrentIndex { get; set; }
+
+        /// <summary>
+        /// Emitted when the searched element is found.
+        /// </summary>
+        [DocumentedByXml]
+        public TEvent Found = new TEvent();
```

Comment:
No properties? Can't remember our decision.

Reply:
There's no benefit making the UnityEvent a property is there? you can't set anything with them and the unity standard is to make them fields.

Reply:
👍